hack/godep-save.sh

```diff
@@ -22,40 +22,65 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 source "${KUBE_ROOT}/hack/lib/util.sh"
 
+TMPPFX="_godep_tmp."
```

Comment:
Suggested patch:
```
diff --git a/hack/godep-save.sh b/hack/godep-save.sh
index 3c04e341ab..400bdd0232 100755
--- a/hack/godep-save.sh
+++ b/hack/godep-save.sh
@@ -22,34 +22,35 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${KUBE_ROOT}/hack/lib/init.sh"
 source "${KUBE_ROOT}/hack/lib/util.sh"
 
-TMPPFX="_godep_tmp."
-
 kube::log::status "Ensuring prereqs"
 kube::util::ensure_single_dir_gopath
 kube::util::ensure_no_staging_repos_in_gopath
-
 kube::util::ensure_godep_version
+kube::util::ensure-temp-dir
+
+GODEP_TMPPFX="${KUBE_TEMP}/_godep_tmp."
+
 
-function cleanup() {
-    if [[ -d "${TMPPFX}vendor" ]]; then
-        kube::log::error "${TMPPFX}vendor exists, restoring it"
+function kube::godep::cleanup() {
+    if [[ -d "${GODEP_TMPPFX}vendor" ]]; then
+        kube::log::error "${GODEP_TMPPFX}vendor exists, restoring it"
         rm -rf vendor
-        mv "${TMPPFX}vendor" vendor
+        mv "${GODEP_TMPPFX}vendor" vendor
     fi
-    if [[ -d "${TMPPFX}Godeps" ]]; then
-        kube::log::error "${TMPPFX}Godeps exists, restoring it"
+    if [[ -d "${GODEP_TMPPFX}Godeps" ]]; then
+        kube::log::error "${GODEP_TMPPFX}Godeps exists, restoring it"
         rm -rf Godeps
-        mv "${TMPPFX}Godeps" Godeps
+        mv "${GODEP_TMPPFX}Godeps" Godeps
     fi
 }
-trap cleanup EXIT
+kube::util::trap_add kube::godep::cleanup EXIT
 
 # Clear old state, but save it in case of error
 if [[ -d vendor ]]; then
-    mv vendor "${TMPPFX}vendor"
+    mv vendor "${GODEP_TMPPFX}vendor"
 fi
 if [[ -d Godeps ]]; then
-    mv Godeps "${TMPPFX}Godeps"
+    mv Godeps "${GODEP_TMPPFX}Godeps"
 fi
 
 # Some things we want in godeps aren't code dependencies, so ./...
@@ -83,4 +84,4 @@ kube::log::status "Updating LICENSES file"
 hack/update-godep-licenses.sh >/dev/null
 
 # Clean up
-rm -rf "${TMPPFX}vendor" "${TMPPFX}Godeps"
+rm -rf "${GODEP_TMPPFX}vendor" "${GODEP_TMPPFX}Godeps"
diff --git a/hack/lib/util.sh b/hack/lib/util.sh
index 2b832335f1..4ddc7bbf99 100755
--- a/hack/lib/util.sh
+++ b/hack/lib/util.sh
@@ -85,7 +85,7 @@ kube::util::trap_add() {
     if [[ -z "${existing_cmd}" ]]; then
       new_cmd="${trap_add_cmd}"
     else
-      new_cmd="${existing_cmd};${trap_add_cmd}"
+      new_cmd="${trap_add_cmd};${existing_cmd}"
     fi
 
     # Assign the test
```

What this does:
- Uses ensure-temp-dir to create a system temp directory that gets cleaned up, out of tree (so that _godep_tmp dirs don't get left behind)
- Uses a more specific variable name
- Fixes ordering in the trap_add function so that more recently added traps will get completed first (such as restoring the vendor dir before the temp dir is cleaned up)

WDYT?

Reply:
KUBE_TEMP is not guaranteed to be on the same device, which could turn the `mv` into a slow `cp`.  We could instead do _output/local/tmp/$RANDOM or something, if you feel strongly.

Will apply parts of your patch and we can discuss the rest.  PTAL